msal/application.py

```diff
@@ -700,6 +700,35 @@ def _validate_ssh_cert_input_data(self, data):
                     "you must include a string parameter named 'key_id' "
                     "which identifies the key in the 'req_cnf' argument.")
 
+    def acquire_token_by_refresh_token(self, refresh_token, scopes):
+        """Acquire token(s) based on a refresh token (RT) obtained from elsewhere.
+
+        In normal MSAL scenarios, you do NOT need to use this method.
+        MSAL maintains RT automatically inside its token cache, and use it to
+        find right tokens for you when you call :func:`~acquire_token_silent`.
+
+        You use this method only when you have old RTs from elsewhere,
+        and now you want to migrate them into MSAL.
+        Calling this method results in new tokens automatically storing into MSAL.
+
+        :param str refresh_token: The old refresh token, as a string.
+
+        :param list scopes:
+            The scopes associate with this old RT.
+            Each scope needs to be in the Microsoft identity platform (v2) format.
+            https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal#scopes-not-resources
```

Comment:
>https://docs.microsoft.com/en-us/azure/active-d [](start = 12, length = 47)

aka.ms link?

Reply:
Instead of creating yet another short link, the next commit will change such naked url into "[Scopes not resources](https://docs.microsoft.com/en-us/azure/active-directory/develop/migrate-python-adal-msal#scopes-not-resources)", then using its canonical long url will not be a problem.